ModelHypothesis[] hypotheses

#initial guess if this can be a correct recognition result at all
bool accept